pepdbagent/modules/annotation.py

```diff
@@ -256,72 +237,163 @@ def _get_projects(
         admin: Union[str, List[str]] = None,
         limit: int = DEFAULT_LIMIT,
         offset: int = DEFAULT_OFFSET,
+        order_by: str = "update_date",
+        order_desc: bool = False,
     ) -> List[AnnotationModel]:
         """
         Get project by providing search string.
         :param namespace: namespace where to search for a project
-        :param search_str: search string that has to be found in the name, tag or project description
+        :param search_str: search string that has to be found in the name or tag
         :param admin: True, if user is admin of the namespace [Default: False]
         :param limit: limit of return results
         :param offset: number of results off set (that were already showed)
+        :param order_by: sort the result-set by the information
+            Options: ["name", "update_date", "submission_date"]
+            [Default: "update_date"]
+        :param order_desc: Sort the records in descending order. [Default: False]
         :return: list of found projects with their annotations.
         """
         _LOGGER.info(
             f"Running annotation search: (namespace: {namespace}, query: {search_str}."
         )
-        if search_str:
-            search_str = f"%%{search_str}%%"
-            search_sql_values = (
-                search_str,
-                search_str,
-                search_str,
-            )
-            search_sql = f"""({NAME_COL} ILIKE %s or ({PROJ_COL}->>'description') ILIKE %s or {TAG_COL} ILIKE %s) and"""
-        else:
-            search_sql_values = tuple()
-            search_sql = ""
 
-        admin_tuple = tuple_converter(admin)
+        if admin is None:
+            admin = []
+        statement = select(
```

Comment:
Would it be better to do something that mimics `SELECT *` ?

Reply:
I would love do do it, but unfortunately, in this table we have project value, and we don't want to return it.

Reply:
it's better to be explicit anyway, SELECT * is almost always a bad idea...

Reply:
Ohh I see. That makes sense. SQL should have `SELECT *` **except** `COLUMN A`!

https://stackoverflow.com/questions/729197/exclude-a-column-using-select-except-columna-from-tablea